### PR TITLE
[Breaking Imports] Refactor MixUp and CutMix into MixHandler

### PR DESCRIPTION
**Pull request by muellerzr**
# **Breaking Changes**

Refactors MixUp and CutMix into a general MixHandler class, with MixUp and CutMix simply implementing a `before_batch` to perform the data augmentation. To use now do `from fastai.callback.mixup import *` to get access to both

### CODE REVIEW

**Comment by jph00** [comment]
What's the breaking change(s) exactly?

**Comment by jph00** [comment]
Let's change the module name back to `mixup` from `mixups` so that's one less breaking change

**Comment by muellerzr** [comment]
Was about to say, the imports would be the major shift/break. Sounds good

**Comment by muellerzr** [comment]
@jph00 should be all good to go now!

**Comment by jph00** [comment]
Are there any breaking changes as far as you know?

Have you trained some models before and after this change and confirmed they're at least as accurate as previously? (If not, please do so! :) )

**Comment by muellerzr** [comment]
Yup, I have been using this in a kaggle competition and they are just as accurate, only breaking changes are the imports (which we've fixed to a degree :) )

**Comment by muellerzr** [comment]
@jph00 can also confirm other Mix variants can be adapted easily, just did `Ricap` a moment ago and it trains as I would expect :) (would Ricap also be a good idea to include in this "mix" module? I can do that as a separate PR later)

**Comment by jph00** [comment]
Yes RICAP would be great.

**Comment by jph00** [comment]
Wonderful! :)

**Comment by muellerzr** [comment]
Great! Will definitely make a PR post this then :)

**Comment by muellerzr** [comment]
I'd like to put this on pause just for a few days, Ignacio Oguiza (from the forums) had a chance to look over the code as well and made some recommendations but I won't be able to reflect them in my code for a little bit.

**Comment by jph00** [comment]
@muellerzr any news on this?

**Comment by muellerzr** [comment]
Not quite yet, sorry! Final projects. I’ll be able to look at this over the
weekend.

**Comment by muellerzr** [comment]
@jph00 okay changes have been pushed. Here's what Ignacio suggested:

Regarding CutMix: 
> I went back through the original CutMix paper and lambda can take any value between 0 and 1, not .5 and 1, so I've changed the logic for that. 
> Also if the input shape is [batch size x channels x height x width], then x (which corresponds to w) needs to be applied to the 3rd dimension, and y (H) to the 2nd dimension. This doesn't make any difference with sqaured images, but it'd break with non-squared images.

Regarding my most recent changes:
In order to not have to worry about too much magic and make this adaptable for Mix that use custom losses, I changed up how `MixHandler` can act. We have a `mixup_criterion` which is for `MixUp` and `CutMix`, and if other loss functions are needed then they can directly pass in their custom ones as `loss_func`

**Comment by jph00** [comment]
Thankyou :)
